### DISCUSSION TITLE
Termostato wifi BEOK Tuya non correttamente riconosciuto

### DISCUSSION THREAD

**Discussion by flayy73** [indomus/forum#796] [HUB: Home Assistant]
Salve a tutti,
sto configurando la mia nuova casa (full domotica con HA), e per la termoregolazione vorrei usare i termostati BEOK 

per l'integrazione in HA uso Tuya v2 che mi riconosce sia il device che l'entity. 
![immagine](https://user-images.githubusercontent.com/86019626/135441039-a67e8947-508b-49ea-b687-c0b509a31165.png)
![immagine](https://user-images.githubusercontent.com/86019626/135441230-63ca985b-8822-4cbe-8c00-c0f4c330d028.png)

ma come si vede, con dei valori completamente sballati rispetto al reale ed inoltre senza la possibilità di accenderlo o di cambiare la temperatura di set.

Sul log HA di tuya2 vedo questo:
***
2021-09-29 19:33:42 DEBUG (SyncWorker_0) [tuya iot] Response: {
  "result": [
    {
      "active_time": 1628280192,
      "biz_type": 18,
      "category": "wk",
      "create_time": 1622049142,
      "icon": "smart/icon/15264583415nnlzwfs1qd2wqz6i937op3nmi_0.png",
      "id": "878608518caab5e502f5",
      "ip": "***",
      "lat": "***",
      "local_key": "***",
      "lon": "***",
      "model": "",
      "name": "Termostato01",
      "online": true,
      "owner_id": "35277597",
      "product_id": "IAYz2WK1th0cMLmL",
      "product_name": "thermostat",
      "status": [
        {
          "code": "switch",
          "value": false
        },
        {
          "code": "temp_set",
          "value": 30
        },
        {
          "code": "upper_temp",
          "value": 47
        },
        {
          "code": "eco",
          "value": false
        },
        {
          "code": "child_lock",
          "value": false
        }
      ],
      "sub": false,
      "time_zone": "+01:00",
      "uid": "***",
      "update_time": 1632846559,
      "uuid": "878608518caab5e502f5"
    }
  ],
  "success": true,
  "t": 1632936822680
}
***

I campi switch, tem_set e upper_temp sono quelli che mi interessano. Ho provato a guardare il climate.py di tuya_v2 ma anche se mastico un po di pyton, per me è un po troppo. Secondo voi è possibile configurare nei custom components un generic thermostat che peschi quelle 3 variabili e me le visualizzi in lovelace? 
Ho provato con la repository programmable_thermostat (https://github.com/custom-components/climate.programmable_thermostat) che va un po meglio, mi mostra la temperatura (anche se raddoppiata) ma niente altro...
Qualche idea/suggerimento su come proseguire?

Grazie

**Comment by LoRacolo** [comment]
Ciao, come sei riuscito ad ottenere il log che hai postato? Io ho fatto una procedura simile, solo che uso come integrazione Tuya, invece che Tuya v2. Ho un device di tipo thermostat che a sua volta è collegato ad una singola entity, di tipo climate. Immagino che ci dovrebbe essere anche un'altra entità di tipo sensor che però non è stata creata: forse è quello il problema.
